san/backtest.py

```diff
@@ -22,40 +22,40 @@ def __init__(self, returns:pd.Series, trades:pd.Series, lagged=True, transaction
             trades.iloc[0] = False
         self.strategy_returns = ((returns * percent_invested_per_trade) * trades)
         self.trades = trades
-        
-        self.nr_trades = 0
+
+        self.nr_trades = {'buy': [], 'sell': []}
         for i in range(1, len(trades)):
             if trades[i] != trades[i - 1]:
                 self.strategy_returns.iloc[i] -= transaction_cost
-                self.nr_trades += 1
+                if trades[i]:
+                    self.nr_trades['buy'].append(self.trades.index[i])
+                else:
+                    self.nr_trades['sell'].append(self.trades.index[i])
         if trades[-1]:  # include last day sell to make benchmark possible
-            self.nr_trades += 1
+            self.nr_trades['sell'].append(self.trades.index[i])
 
-        self.performance = ((self.strategy_returns * percent_invested_per_trade) * trades + 1).cumprod() - 1
+        self.performance = ((self.strategy_returns * percent_invested_per_trade) * self.trades + 1).cumprod() - 1
         self.benchmark = (returns + 1).cumprod() - 1
-        self.sharpe_ratio = (self.strategy_returns.mean() * 365) / (self.strategy_returns.std() * np.sqrt(365))
-        
-        running_value = np.array(self.performance)
-        running_value[0] = 0
-        a = np.argmax(np.maximum.accumulate(running_value) - running_value)  # end of the period
-        b = np.argmax(running_value[:a])  # start of period
-        self.maximum_drawdown = ((running_value[a] - running_value[b]) / running_value[b]) * 100  # Maximum Drawdown
-
-
-    def get_sharpe_ratio(self):
-        return round(self.sharpe_ratio, 2)
 
+    def get_sharpe_ratio(self, decimals=2):
+        sharpe_ratio = (self.strategy_returns.mean() * 365) / (self.strategy_returns.std() * np.sqrt(365))
```

Comment:
should this always be 365 or it should be configurable?

Reply:
Its annualizing the sharpe ratio. The issue with this is that it only works for daily data of course. We could try some mechanism that figures out the granularity and then annualizes accordingly. I think pandas has something for this. But I guess this would also make it more fragile since the entered dataframe needs precisely specified datetime...